pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -529,3 +529,187 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
 		}
 	}
 }
+
+func TestConvertRbacRulesToFilterConfigPermissive(t *testing.T) {
+	roles := []model.Config{
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-1"},
+			Spec:       generateServiceRole([]string{"service"}, []string{"GET"}),
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-2"},
+			Spec:       generateServiceRole([]string{"service"}, []string{"POST"}),
+		},
+	}
+	bindings := []model.Config{
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-1"},
+			Spec:       generateServiceBinding("user1", "service-role-1", rbacproto.EnforcementMode_ENFORCED),
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-2"},
+			Spec:       generateServiceBinding("user2", "service-role-1", rbacproto.EnforcementMode_PERMISSIVE),
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-3"},
+			Spec:       generateServiceBinding("user3", "service-role-2", rbacproto.EnforcementMode_ENFORCED),
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-4"},
+			Spec:       generateServiceBinding("user4", "service-role-2", rbacproto.EnforcementMode_PERMISSIVE),
+		},
+	}
+
+	enforcePolicy1 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "GET")},
+		Principals:  []*policy.Principal{generatePrincipal("user1")},
+	}
+
+	enforcePolicy2 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "POST")},
+		Principals:  []*policy.Principal{generatePrincipal("user3")},
+	}
+
+	permissivePolicy1 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "GET")},
+		Principals:  []*policy.Principal{generatePrincipal("user2")},
+	}
+
+	permissivePolicy2 := &policy.Policy{
+		Permissions: []*policy.Permission{generatePermission(":method", "POST")},
+		Principals:  []*policy.Principal{generatePrincipal("user4")},
+	}
+
+	enforcedRbac := &policy.RBAC{
+		Action: policy.RBAC_ALLOW,
+		Policies: map[string]*policy.Policy{
+			"service-role-1": enforcePolicy1,
+			"service-role-2": enforcePolicy2,
+		},
+	}
+	shadowRbac := &policy.RBAC{
+		Action: policy.RBAC_ALLOW,
+		Policies: map[string]*policy.Policy{
+			"service-role-1": permissivePolicy1,
```

Comment:
[optional] I would define these policies and permission inline here, to avoid adding too many variables, also quicker to cross check.